src/Utils/Responsive/Media.tsx

```diff
@@ -0,0 +1,268 @@
+import PropTypes from "prop-types"
+import React from "react"
+import styled, { css, InterpolationValue } from "styled-components"
+
+const MutuallyExclusiveProps = [
+  "query",
+  "at",
+  "lessThan",
+  "greaterThan",
+  "greaterThanOrEqual",
+  "between",
+  "interaction",
+]
+
+// TODO: All of these props should be mutually exclusive. Using a union should
+//       probably be made possible by https://github.com/Microsoft/TypeScript/pull/27408.
+export interface Props<B, I> {
+  not?: boolean
+
+  /**
+   * Allows you to pass in any CSS Media Query that will be used to
+   * conditionally show or hide the children.
+   *
+   * Use this sparingly and consider making your use-case part of the API,
+   * because these raw strings cannot be used directly on React Native.
+   */
+  query?: string
+
+  /**
+   * Children will only be shown if the viewport matches the specified
+   * breakpoint. That is, a viewport width that’s higher than the configured
+   * breakpoint value, but lower than the value of the next breakpoint, if any
+   * larger breakpoints exist.
+   *
+   * @example
+
+     ```tsx
+     const { Media } = createMedia({
+       breakpoints: {
+         xs: 0,
+         sm: 768,
+         md: 1024,
+       },
+       interactions: {},
+     })
+ 
+     // Matches a viewport that has a width between 0 and 768
+     <Media at="xs">ohai</Media>
+ 
+     // Matches a viewport that has a width between 768 and 1024
+     <Media at="sm">ohai</Media>
+ 
+     // Matches a viewport that has a width over 1024
+     <Media at="md">ohai</Media>
+     ```tsx
+   *
+   */
+  at?: B
+
+  lessThan?: B
+  greaterThan?: B
+  greaterThanOrEqual?: B
+  between?: [B, B]
+  interaction?: I
+  children: React.ReactNode | RenderProp
+}
+
+type RenderProp = ((
+  generatedStyle: RenderPropStyleGenerator
+) => React.ReactNode)
+
+type RenderPropStyleGenerator = (
+  matchingStyle?: InterpolationValue[]
+) => InterpolationValue[]
+
+/**
+ * breakpoints are 0-based
+ */
+export function createMedia<
+  C extends {
+    breakpoints: { [key: string]: number }
+    interactions: { [key: string]: (negated: boolean) => string }
+  }
+>(config: C) {
+  type Breakpoint = keyof C["breakpoints"]
+  type Interaction = keyof C["interactions"]
+
+  const MediaContext = React.createContext<{ onlyRender?: Breakpoint[] }>({})
+  const MediaContextProvider: React.SFC<{ onlyRender?: Breakpoint[] }> = ({
+    onlyRender,
```

Comment:
Can you add a comment that describes this prop a bit?

Reply:
Yup yup. I’m now also thinking to rename it to `onlyRenderAt`, we can iterate and a separate one for interactions later on when we actually start using this on the server.